providers/src/airflow/providers/standard/triggers/file.py

```diff
@@ -69,9 +69,11 @@ async def run(self) -> typing.AsyncIterator[TriggerEvent]:
                     mod_time = datetime.datetime.fromtimestamp(mod_time_f).strftime("%Y%m%d%H%M%S")
                     self.log.info("Found File %s last modified: %s", path, mod_time)
                     yield TriggerEvent(True)
+                    await asyncio.sleep(self.poke_interval)
```

Comment:
Won't this slow down the throughput from the triggerer, as we have this sleep that must be performed before the slot frees up for use again?

You could probably exacerbate/demonstrate this by setting a high poke interval and a triggerer that only runs a single trigger at a time.

Reply:
Since it is an async sleep that will not block the triggerer. The triggerer will use this time to run other async operations. At least that's my understanding

Reply:
It might be worth testing to be sure, but I'd also expect the triggerer to move on to another trigger while waiting for this sleep. Otherwise, the poke sleeps themselves would block the triggerer massively.

Reply:
(Not important now that we are going a different direction, but for posterity)
The main thread won't be blocked, but if you are at capacity, I think you'd see the triggerer is unable to start another trigger until the sleeping one wraps up.